src/frequenz/sdk/timeseries/battery_pool/_component_metric_fetcher.py

```diff
@@ -128,7 +140,9 @@ async def fetch_next(self) -> ComponentMetricsData | None:
             return None
         except asyncio.TimeoutError:
             # Next time wait infinitely until we receive any message.
-            _logger.debug("Component %d stopped sending data.", self._component_id)
+            _missing_data_logger.debug(
+                "Component %d stopped sending data.", self._component_id
```

Comment:
I would rephrase because when it is repeated, one might think that it **just** stopped sending data now. Also since we have instances where the problem was in the data pipeline and not the component itself, maybe is more accurate to say we are not receiving data.

```suggestion
                "No data received for component %d.", self._component_id
```